vignettes/server-mode.Rmd

```diff
@@ -0,0 +1,396 @@
+---
+title: "Server Mode Guide for bettr"
+author: "bettr package authors"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Server Mode Guide for bettr}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+
+**Version 1.6.0 and later**
+
+## Table of Contents
+
+1.  [Overview](#overview)
+2.  [What's New in Version 1.6.0](#whats-new-in-version-160)
+3.  [Server Mode Basics](#server-mode-basics)
+4.  [JSON Serialization](#json-serialization)
+5.  [URL Parameters for Programmatic
+    Loading](#url-parameters-for-programmatic-loading)
+6.  [Browser Cache Management](#browser-cache-management)
+
+------------------------------------------------------------------------
+
+## Overview {#overview}
+
+Version 1.6.0 introduces **server mode**, a new way to deploy and use
+bettr that enables browser-based JSON file uploads, automatic caching,
+and programmatic data loading via URLs. This makes bettr suitable for:
+
+-   **Multi-user server deployments** where different users need to
+    visualize their own data
+-   **Web-based workflows** without requiring R programming knowledge
+-   **Data sharing** through portable JSON files and direct links
+
+------------------------------------------------------------------------
+
+## What's New in Version 1.6.0
+
+### 1. Server Mode for JSON Uploads
+
+The `bettr()` function now supports a `serverMode` parameter that
+launches the app in a special mode where users upload JSON files through
+the browser.
+
+**Benefits:**
+
+-   No R programming required for end users
+-   Multiple users can use the same server instance with their own data
+-   Data is isolated per user session
+-   Automatic browser caching for quick reload
+
+### 2. JSON Serialization {#json-serialization}
+
+New functions for converting bettr data to/from JSON format:
+
+-   `bettrToJSON()` - Export `SummarizedExperiment` to JSON
+-   `bettrFromJSON()` - Import JSON back to `SummarizedExperiment`
```

Comment:
maybe worth at this spot or somewhere around here to show how the json file looks like?
Some simple printout of that, possibly inside a details-summary tag group to avoid this taking too much space, in case this gets wordy